packages/ember-glimmer/lib/syntax/curly-component.js

```diff
@@ -102,17 +102,17 @@ class CurlyComponentManager {
 
     assert('You cannot use `classNameBindings` on a tag-less component: ' + component.toString(), () => {
       let { classNameBindings, tagName } = component;
-      return tagName || !classNameBindings || classNameBindings.length === 0;
+      return tagName !== '' || !classNameBindings || classNameBindings.length === 0;
     });
 
     assert('You cannot use `elementId` on a tag-less component: ' + component.toString(), () => {
       let { elementId, tagName } = component;
-      return tagName || (!elementId && elementId !== '');
+      return tagName !== '' || (!elementId && elementId !== '');
```

Comment:
This is causing some tests to randomly fail with this assertion. The default `tagName` for all components is intentionally `null`, and that gets converted into a 'div' later down the road. I don't know how this is not causing a lot more tests to fail – it might indicate a deeper problem, e.g. we are accidentally leaking `tagName: '...'` onto the prototype, etc.